frontend/packages/knative-plugin/src/utils/knative-topology-utils.ts

```diff
@@ -185,53 +196,20 @@ export const getTrafficTopologyEdgeItems = (resource: K8sResourceKind, { data })
   return edges;
 };
 
-/**
- * Form Group data for service sources
- */
-export const getTopologyServiceGroupItems = (
-  resource: K8sResourceKind,
-  groups: Group[],
-): Group[] => {
-  const labels = _.get(resource, ['spec', 'template', 'metadata', 'labels'], []);
-  const uid = _.get(resource, ['metadata', 'uid']);
-  _.forEach(labels, (label, key) => {
-    if (key !== 'app.kubernetes.io/part-of') {
-      return;
-    }
-    // find and add the groups
-    const groupExists = _.some(groups, {
-      name: label,
-    });
-    if (!groupExists) {
-      groups.push({
-        id: `group:${label}`,
-        name: label,
-        nodes: [uid],
-      });
-    } else {
-      const gIndex = _.findIndex(groups, { name: label });
-      groups[gIndex].nodes.push(uid);
-    }
-  });
-  return groups;
-};
-
 /**
  * Filter Knative resources based on part-of label
  */
 export const filterKnativeBasedOnActiveApplication = (
   data: K8sResourceKind[],
+  type: string,
   application: string,
 ): K8sResourceKind[] => {
   const PART_OF = 'app.kubernetes.io/part-of';
-  if (!application) {
+  if (!application || type === nodeType.Revision) {
```

Comment:
Why are we adding this check for nodeType?

Reply:
added as `revisions` shouldn't be associated with any group irrespective of if it has `part-of`  or not. Revisions should only be associated with service